File: app/src/main/java/org/mozilla/focus/utils/StatusBarUtils.java
package org.mozilla.focus.utils;

import android.view.View;
import android.view.WindowInsets;

/**
 * Created by Fer on 08/03/2018.
 */

public class StatusBarUtils {
    public static int STATUS_BAR_SIZE = -1;

    public interface StatusBarHeightListener {
        void onStatusBarHeightFetched(int statusBarHeight);
    }

    public static void getStatusBarHeight(View view, final StatusBarHeightListener listener) {
        if (STATUS_BAR_SIZE > 0)
            listener.onStatusBarHeightFetched(STATUS_BAR_SIZE);

        view.setOnApplyWindowInsetsListener(new View.OnApplyWindowInsetsListener() {
            @Override
            public WindowInsets onApplyWindowInsets(View v, WindowInsets insets) {
                STATUS_BAR_SIZE = insets.getSystemWindowInsetTop();
                listener.onStatusBarHeightFetched(STATUS_BAR_SIZE);
                return insets;
            }
        });
    }
}

File: app/src/main/java/org/mozilla/focus/utils/StatusBarUtils.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.focus.utils

import android.view.View

object StatusBarUtils {
    private var STATUS_BAR_HEIGHT = -1

    interface StatusBarHeightListener {
        fun onStatusBarHeightFetched(statusBarHeight: Int)
    }

    fun getStatusBarHeight(view: View?, listener: StatusBarHeightListener) {
        if (STATUS_BAR_HEIGHT > 0) {
            listener.onStatusBarHeightFetched(STATUS_BAR_HEIGHT)
        }

        view!!.setOnApplyWindowInsetsListener { _, insets ->
            STATUS_BAR_HEIGHT = insets.systemWindowInsetTop
            listener.onStatusBarHeightFetched(STATUS_BAR_HEIGHT)
            insets
        }
    }
}
